ci/pingcap_tidb_operator_build_kind.groovy

```diff
@@ -27,22 +27,61 @@ def call(BUILD_BRANCH, CREDENTIALS_ID) {
 				def WORKSPACE = pwd()
 				dir("${PROJECT_DIR}"){
 					stage('build tidb-operator binary'){
-						checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: "${BUILD_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: "${CREDENTIALS_ID}", refspec: '+refs/pull/*:refs/remotes/origin/pr/*', url: "${BUILD_URL}"]]]
+						if( BUILD_BRANCH ==~ /[a-z0-9]{40}/ ){
+							// checkout pull request
+							checkout changelog: false,
+							poll: false,
+							scm: [
+								$class: 'GitSCM',
+								branches: [[name: "${BUILD_BRANCH}"]],
+								doGenerateSubmoduleConfigurations: false,
+								extensions: [],
+								submoduleCfg: [],
+								userRemoteConfigs: [[
+									credentialsId: "${CREDENTIALS_ID}",
+									refspec: '+refs/pull/*:refs/remotes/origin/pr/*',
+									url: "${BUILD_URL}",
+								]]
+							]
+						} else {
+							// checkout branch, such as: master、release-1.0、release-1.1
+							checkout changelog: false,
+							poll: false,
+							scm: [
+								$class: 'GitSCM',
+								branches: [[name: "${BUILD_BRANCH}"]],
+								doGenerateSubmoduleConfigurations: false,
+								extensions: [],
+								submoduleCfg: [],
+								userRemoteConfigs: [[
+									credentialsId: "${CREDENTIALS_ID}",
+									url: "${BUILD_URL}",
+								]]
+							]
+						}
 						//git credentialsId: "k8s", url: "${BUILD_URL}", branch: "${ghprbActualCommit}"
```

Comment:
how about remove this commented line in this PR too?

Reply:
Done